pgxn/neon/file_cache.c

```diff
@@ -1567,8 +1567,12 @@ local_cache_pages(PG_FUNCTION_ARGS)
 				hash_seq_init(&status, lfc_hash);
 				while ((entry = hash_seq_search(&status)) != NULL)
 				{
-					for (int i = 0; i < BLOCKS_PER_CHUNK; i++)
-						n_pages += GET_STATE(entry, i) == AVAILABLE;
+					/* Skip hole tags */
+					if (NInfoGetRelNumber(BufTagGetNRelFileInfo(entry->key)) != 0)
```

Comment:
A few catalogs have relfilenumber = 0 in pg_class, and use the "relfilenumber map" facility to track the relfilenumber:
```
neondb=> select relname from pg_class where relfilenode = 0 and relkind='r';
        relname        
-----------------------
 pg_type
 pg_authid
 pg_subscription
 pg_attribute
 pg_proc
 pg_class
 pg_database
 pg_db_role_setting
 pg_tablespace
 pg_auth_members
 pg_shdepend
 pg_shdescription
 pg_replication_origin
 pg_shseclabel
 pg_parameter_acl
(15 rows)
```

I presume we have looked up the relfilenumber for these before reaching the LFC code, so that this doesn't get fooled by them?

Reply:
Yes, certainly - for catalog relation filenodemap is used.
Otherwise everything will not work:)
There is check in `smgropen`:
```
	Assert(RelFileNumberIsValid(rlocator.relNumber));
```